projects/packages/password-checker/tests/php/test-class-password-checker.php

```diff
@@ -0,0 +1,125 @@
+<?php //phpcs:ignore WordPress.Files.FileName.InvalidClassFileName
+
+/**
+ * Tests the Password_Checker package.
+ *
+ * @package automattic/jetpack-password-checker
+ */
+
+namespace Automattic\Jetpack;
+
+use WorDBless\BaseTestCase;
+
+/**
+ * Test Password_Checker class
+ */
+class Password_Checker_Test extends BaseTestCase {
+	/**
+	 * User ID.
+	 *
+	 * @var int|\WP_Error
+	 */
+	private $user_id;
+
+	/**
+	 * User object.
+	 *
+	 * @var \WP_User
+	 */
+	private $user;
+
+	/**
+	 * Password_Checker object.
+	 *
+	 * @var Password_Checker
+	 */
+	private $password_checker;
+
+	/**
+	 * Initialize tests.
+	 *
+	 * @before
+	 */
+	public function set_up() {
+		$this->user_id = wp_insert_user(
+			array(
+				'user_login' => 'test-user',
+				'user_pass'  => '123',
+				'first_name' => 'Test',
+				'last_name'  => 'User',
+				'nickname'   => 'test',
+				'role'       => 'subscriber',
+			)
+		);
+
+		$this->user = new \WP_User( $this->user_id );
+
+		$this->password_checker = new Password_Checker( $this->user );
+	}
+
+	/**
+	 * Test the password checker.
+	 *
+	 * @dataProvider rule_provider
+	 *
+	 * @param string $rule            Rule name.
+	 * @param string $password        The password.
+	 * @param bool   $expected_result The expected result.
+	 * @param string $output_message  The output message.
+	 */
+	public function test_password( $rule, $password, $expected_result, $output_message ) { // phpcs:ignore VariableAnalysis.CodeAnalysis.VariableAnalysis.UnusedVariable
+		$tests = apply_filters( 'password_checker_tests', array() ); // phpcs:ignore VariableAnalysis.CodeAnalysis.VariableAnalysis.UnusedVariable
+
+		$results = $this->password_checker->test( $password, true );
+		$this->assertSame( $expected_result, $results['passed'] );
+	}
+
+	/**
+	 * Data provider for password tests.
+	 *
+	 * @return array
+	 */
+	public function rule_provider() {
+		/**
+		 * Data format.
+		 *
+		 * Param 1 -> rule
+		 * Param 2 -> password
+		 * Param 3 -> expected_result
+		 * Param 4 -> output_message
+		 */
+
+		return array(
+			'no_backslashes'   => array(
```

Comment:
@anomiex is there a standard way to pass the array key like 'no_backslashes' to the test method?
I was hoping I could avoid the duplicate data here.

The first param to the test method is the $rule name that needs to be tested. Is there a way to omit that and pass the associative array key instead?

Reply:
Not that I know of.

You might well want to have multiple tests for a single rule anyway, which would need different array keys.

Reply:
Agree with Brad we'll want multiple tests for various rules if we expand this package at a later date.